src/plugins/expressions/common/execution/execution.ts

```diff
@@ -45,6 +45,21 @@ import { ExpressionExecutionParams } from '../service';
 import { TablesAdapter } from '../util/tables_adapter';
 import { ExpressionsInspectorAdapter } from '../util/expressions_inspector_adapter';
 
+/**
+ * The result returned after an expression function execution.
+ */
+export interface ExecutionResult<Output> {
+  /**
+   * Partial result flag.
+   */
+  partial: boolean;
```

Comment:
We need more than just a flag for partial data, we need some kind of status indicator as well. Typically I'd imagine this is the shard responses: 100 total shards, 5 successful, 50 skipped, 1 failure.

Reply:
We cannot predict that because an expression function can emit an indefinite number of results. So the best you can do is to implement the progress indication in a function.

Reply:
@dokmic 

> So the best you can do is to implement the progress indication in a function

Can you elaborate on how this would look like? I think this is an important part of supporting partial results.

Reply:
@dokmic Elasticsearch indicates the number of successful vs total shards on every partial status call, this is what I'm referring to:  [async search status](https://www.elastic.co/guide/en/elasticsearch/reference/current/async-search.html#get-async-search-status)

Reply:
@flash1293 @wylieconlon this PR does not try to address progress indication apart from letting the consumer know when result is partial vs final. Progress reporting will be added at a later time, right now its not even supported on our low level search service.

Reply:
@ppisljar Right, I understand that it's not supported yet. What I'm asking is that we implement an API that _can_ support a progress indicator, not just a boolean.

Reply:
we didn't want to do anything more for now as we don't know yet in what form progress from es will be provided to us or how we'd want to report this on expression level. in the future we are likely to keep this partial flag and add another progress variable.